docs/.hooks/expand_blocks.py

```diff
@@ -40,21 +39,9 @@ def _config_example_replace(m):
 """
 
 
-def on_config(
-    config,
-    **kwargs,  # noqa: ARG001
-):
-    config.markdown_extensions.append(MyExtension())
-
-
-class MyExtension(Extension):
-    def extendMarkdown(self, md):  # noqa: N802, PLR6301
-        md.preprocessors.register(MyPreprocessor(), 'mypreprocessor', 100)
-
-
-class MyPreprocessor(Preprocessor):
+class ExpandedBlocksPreprocessor(Preprocessor):
     def run(self, lines):  # noqa: PLR6301
         markdown = '\n'.join(lines)
         markdown = _config_example_regex.sub(_config_example_replace, markdown)
         markdown = _code_tab_regex.sub(_code_tab_replace, markdown)
-        return markdown.split('\n')
+        return markdown.splitlines()
```

Comment:
`splitlines` actually achieves a slightly different outcome - this drops a newline at the end
https://docs.python.org/3/library/stdtypes.html#str.splitlines